api/src/test/java/org/apache/iceberg/stats/TestStatsUtil.java

```diff
@@ -0,0 +1,146 @@
+  public void contentStatsForSimpleSchema() {
+    Schema schema =
+        new Schema(
+            required(0, "i", Types.IntegerType.get()),
+            required(2, "f", Types.FloatType.get()),
+            required(4, "s", Types.StringType.get()),
+            required(6, "b", Types.BooleanType.get()),
+            required(250_000, "u", Types.UUIDType.get()));
+    Schema expectedStatsSchema =
+        new Schema(
+            optional(
+                146,
+                "content_stats",
+                Types.StructType.of(
+                    optional(10000, "0", fieldStatsFor(Types.IntegerType.get(), 10001)),
+                    optional(10400, "2", fieldStatsFor(Types.FloatType.get(), 10401)),
+                    optional(10800, "4", fieldStatsFor(Types.StringType.get(), 10801)),
+                    optional(11200, "6", fieldStatsFor(Types.BooleanType.get(), 11201)),
+                    optional(50010000, "250000", fieldStatsFor(Types.UUIDType.get(), 50010001)))));
+    Schema statsSchema = new Schema(StatsUtil.contentStatsFor(schema));
+    assertThat(statsSchema.asStruct()).isEqualTo(expectedStatsSchema.asStruct());
+  }
+
+  @Test
+  public void contentStatsForComplexSchema() {
+    Schema schema =
+        new Schema(
+            required(0, "i", Types.IntegerType.get()),
+            required(2, "list", Types.ListType.ofOptional(3, Types.IntegerType.get())),
+            required(
+                6,
+                "simple_struct",
+                Types.StructType.of(
+                    optional(7, "int", Types.IntegerType.get()),
+                    optional(8, "string", Types.StringType.get()))),
+            required(
+                20,
+                "b",
+                Types.MapType.ofOptional(22, 24, Types.IntegerType.get(), Types.StringType.get())),
+            required(30, "variant", Types.VariantType.get()),
+            required(250_000, "u", Types.UUIDType.get()));
+    Schema expectedStatsSchema =
+        new Schema(
+            optional(
+                146,
+                "content_stats",
+                Types.StructType.of(
+                    optional(10000, "0", fieldStatsFor(Types.IntegerType.get(), 10001)),
+                    optional(10600, "3", fieldStatsFor(Types.IntegerType.get(), 10601)),
+                    optional(11400, "7", fieldStatsFor(Types.IntegerType.get(), 11401)),
+                    optional(11600, "8", fieldStatsFor(Types.StringType.get(), 11601)),
+                    optional(14400, "22", fieldStatsFor(Types.IntegerType.get(), 14401)),
+                    optional(14800, "24", fieldStatsFor(Types.StringType.get(), 14801)),
+                    optional(50010000, "250000", fieldStatsFor(Types.UUIDType.get(), 50010001)))));
+    Schema statsSchema = new Schema(StatsUtil.contentStatsFor(schema));
+    assertThat(statsSchema.asStruct()).isEqualTo(expectedStatsSchema.asStruct());
+  }
+
+  private Type fieldStatsFor(Type type, int id) {
+    int fieldId = id;
```

Comment:
I'm generally worried about test code which mimics the implementation. I think ideally we would have some code for reading the stats out of the structure and we could check they matched the values we put in. That may be outside the scope of this file though ...